### DISCUSSION TITLE
Can we talk about Github?

### DISCUSSION THREAD

**Discussion by joaoeliandro** [community/community#18275] [Other Feature Feedback, Questions, & Ideas] [answered]
### Tools
Why use github actions instead of another tool like Jenkins?

**Comment by devcrafta** [comment]
The first and foremost plus point of GitHub Actions over Jenkins, in my opinion, is the ease of setup in GitHub Actions. GitHub Actions operate in the cloud. You also have the option of running it locally, which is called a runner.

Jenkins creates workflows using Declarative Pipelines, which are similar to GitHub Actions workflow files. Jenkins uses stages to run a collection of steps, while GitHub Actions uses jobs to group one or more steps or individual commands. Jenkins and GitHub Actions support container-based builds.

GitHub Actions makes it easy to automate all your software workflows.

Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency. On the other hand, Jenkins is detailed as "An extendable open source continuous integration server".

**Comment by Jannchie** [comment]
If you use Jenkins, you need a server to deploy it. This invariably adds a lot of maintenance and server costs.

The integration between Github Action and Github is more natural than other CI tools, such as Travis CI, so I prefer to use Github Action.

**Comment by MrKrishnaAgarwal** [comment, marked as answer]
GitHub Actions makes it easy to automate all your software workflows, now with world-class CI/CD. Build, test, and deploy your code right from GitHub. Make code reviews, branch management, and issue triaging work the way you want.

Kick off workflows with GitHub events like push, issue creation, or a new release. Combine and configure actions for the services you use, built and maintained by the community.

Whether you want to build a container, deploy a web service, or automate welcoming new users to your open source projects—there's an action for that. Pair GitHub Packages with Actions to simplify package management, including version updates, fast distribution with our global CDN, and dependency resolution, using your existing GITHUB_TOKEN.

GitHub Actions connects all of your tools to automate every step of your development workflow. Easily deploy to any cloud, create tickets in Jira, or publish a package to npm.

Want to venture off the beaten path? Use the millions of open source libraries available on GitHub to create your own actions. Write them in JavaScript or create a container action—both can interact with the full GitHub API and any other public API.

GitHub Actions empowers teams to define highly advanced pipelines without losing the focus on simplicity. It brings the power of the pipeline as code, matrix builds, and predefined environment together in one place. GitHub Actions makes it possible to build and share workflows via reusable steps and delivers what’s required to build the next-generation CI/CD pipeline.

Jenkins and GitHub Actions share some similarities in workflow configuration: Jenkins creates workflows using Declarative Pipelines, which are similar to GitHub Actions workflow files. Jenkins uses stages to run a collection of steps, while GitHub Actions uses jobs to group one or more steps or individual commands.
